### PR TITLE
beta 4 + Copy helpers from pynetworktables util

### PR DESCRIPTION
**Pull request by virtuald**

### CODE REVIEW

File: ntcore/util.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+        self.ntvalue.setValue(self.mkv(value))
+
+
+def ntproperty(
+    key: str,
+    defaultValue,
+    writeDefault: bool = True,
+    doc: str = None,
+    persistent: bool = False,
+    *,
```
Comment (auscompgeek) [inline ntcore/util.py:90]: Since this is technically a new API, should we make these kw-only?

File: ntcore/util.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
+import threading
+import weakref
+
+from typing import Callable, Dict, Optional, Sequence
+
+from ._ntcore import NetworkTableInstance, NetworkTableEntry, Value
+
+__all__ = ["ntproperty", "ChooserControl"]
+
+NetworkTables = NetworkTableInstance.getDefault()
```
Comment (auscompgeek) [inline ntcore/util.py:10]: unused?

```suggestion
```

File: ntcore/util.py (hunk 1 above)
Comment (virtuald) [reply ntcore/util.py:90]: Good point. Going to leave key/defaultValue as positional though.

File: ntcore/src/py2value.cpp [hunk 3]
```diff
@@ -156,21 +156,21 @@ py::cpp_function valueFactoryByType(nt::NetworkTableType type) {
   case nt::NetworkTableType::kString:
     return py::cpp_function([](std::string_view v) { return nt::Value::MakeString(v); });
   case nt::NetworkTableType::kRaw:
-    return py::cpp_function([](std::string_view v) { return nt::Value::MakeString(v); });
+    return py::cpp_function([](std::span<const uint8_t> v) { return nt::Value::MakeRaw(v); });
   case nt::NetworkTableType::kBooleanArray: 
     return py::cpp_function([](std::span<const bool> v) { return nt::Value::MakeBooleanArray(v); });
   case nt::NetworkTableType::kDoubleArray: 
-    return py::cpp_function([](std::span<const double> v) { return nt::Value::MakeDoubleArray(v); });
+    return py::cpp_function([](std::vector<double> v) { return nt::Value::MakeDoubleArray(v); });
```
Comment (auscompgeek) [inline ntcore/src/py2value.cpp:163]:
```suggestion
    return py::cpp_function([](std::vector<double> v) { return nt::Value::MakeDoubleArray(std::move(v)); });
```

**Pull request merged**